pkg/daemon/metrics.go

```diff
@@ -5,11 +5,55 @@ import (
 	"net/http"
 
 	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-// DefaultBindAddress is the port for the metrics listener
-var DefaultBindAddress = ":8797"
+var (
+	// DefaultBindAddress is the port for the metrics listener
+	DefaultBindAddress = ":8797"
+
+	// RebootCounter is a metric incremented when MCD reboots
+	RebootCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "mcd_reboots",
+			Help: "A metric incremented when MCD reboots",
+		}, []string{"node"})
+
+	// DrainCounter is a metric incremented when MCD drains
+	DrainCounter = prometheus.NewGaugeVec(
```

Comment:
this is amazing and opens up to better debugging :pray: another cool one would be an os upgrade one to check on upgrades